_posts/2017-02-05-Retrospective-Swift-at-Artsy.md

```diff
@@ -0,0 +1,337 @@
+
+It's worth highlighting that all of this is done on GitHub, in the open. We can write issues, get responses, and have direct line to the people who are working on something we depend on. This is a stark contrast to the Radar system used internally at Apple, and which external developers have write-only access to. For external contributors radar is opaque, and [totally feels like a waste of time][tnw-radar]. On the other hand, a GitHub issue doesn't have to have to wait for the repo maintainers, others can get value from it and it's publicly indexed. If we had put all our effort into Radars instead of [issues like][eigen_launch] this, the whole community would be worse off.
+
+This isn't all doom and gloom. With Swift the language, and SwiftPM the package manager, Apple are more open with the feedback cycle using tools like Slack, Mailing Lists, JIRA and Twitter.
+
+One aspect of working with JavaScript that has been particularly pleasant is the idea that your language is effectively a buffet. If you want to use the latest features of the language you can opt-in to it. We've slowly added language features, while retaining backwards compatibility. First using [Babel][babel-site], then [Flow][flow-site] and finally with [TypeScript][typescript-site]. 
+
+In contrast, and this may be the last major time it happens, but people refer to the time it took to migrate [in][weeks1] [the][weeks2] [scale][weeks3] [of][weeks4] _weeks_ during the Swift 2 -> 3 migration. Having the language evolve is great, sometimes in ways that you [agree with][swift-api] and sometimes in ways [you don't][closed]. Being able to use your own version of your tools frees you to make it work for you and your business. We have been talking about [extending TypeScript][extend-ts] specifically for our applications.
+
+# React Native, one year later
+
+In our announcement we talked about the lack of nuanced post-mortems on React Native. We're now a year in, we can at least try to help out in that space. We're sticking with React Native for the foreseeable future. It would take some _drastic_ changes in the Apple ecosystem for us to re-consider this decision. So here's the summary after 1 year.
+
+* We can share concepts with web
+* Tools are built for our apps like ours
+* To do it right requires engineers willing to dive deep in JS
+* You need native experience to have a polished app
+* Dependency stack is still obscenely big
+* Opens native engineers to more projects, makes yours more welcoming to others
+* Problems do, and will occur, but everything is fixable by forking
+* Extensive communication with native code gets tricky to test and maintain
+* We ended up re-using quite a lot of existing native code
+* It makes working in native code feel like a bad chore for someone to do
+* Spending so much time in another environment will erode native knowledge
+* We're not going to re-write other existing Apps into React Native
```

Comment:
Reading this on its own feels like it might be a negative judgement on React Native.